server/src/main/java/org/elasticsearch/index/mapper/FieldValueFetcher.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.elasticsearch.index.fielddata.FormattedDocValues;
+
+import java.util.Map;
+
+/**
+ * Interface for retrieving field values by reading field data
+ */
+public interface FieldValueFetcher {
```

Comment:
If it's in `server` it's nice to have a very good name. Its a `FormattedDocValuesFetcher` or something. I dunno. Probably not worth changing it now. If I can't figure out a good way to move it back to the module then we should rename it.